kombu/transport/redis.py

```diff
@@ -1047,6 +1060,39 @@ def _queue_bind(self, exchange, routing_key, pattern, queue):
                                        pattern or '',
                                        queue or '']))
 
+    def _maybe_update_queues_expire(self, queue):
+        """Update expiration on queue keys.
+
+        For each queue, set expiration time in milliseconds.
+        Will only be set if x-expires argument was provided when creating the queue.
+        """
+        if not self._expires:
+            return
+
+        with self.conn_or_acquire() as client:
+            with client.pipeline() as pipe:
+                for pri in self.priority_steps:
+                    pipe = pipe.pexpire(self._q_for_pri(queue, pri), self._expires)
+                pipe.execute()
+
+    def _get_queue_expire(self, args):
+        """Get expiration header named `x-expires` of queue definition.
+
+        Returns expiration time in milliseconds or None if not set.
+
+        Arguments:
+        ---------
+            args (dict): Queue arguments dictionary
+        """
+        try:
+            value = args['arguments']['x-expires']
+            return int(value)
+        except (KeyError, TypeError, ValueError):
+            return None
+
+    def prepare_queue_arguments(self, arguments, **kwargs):
+        return to_rabbitmq_queue_arguments(arguments, **kwargs)
```

Comment:
should we just return a rabbitmq function or think something else?

Reply:
it's rather canonical ampq view of queue_arguments, also mongodb uses exactly the same setup so I assumed it should be fine for redis as well